### PR TITLE
Gql bb v2

### PR DESCRIPTION
**Pull request by asmab89**

### CODE REVIEW

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/GqlConst.kt [hunk 1]
```diff
@@ -53,7 +53,15 @@ object GqlConst {
...
     const val ROOT = "root"
+    const val QUERY_ROOT = "query_root"
+    const val ROOTQUERY = "rootquery"
+    const val ROOTQUERYTYPE = "rootquerytype"
+    const val DOTAQUERY = "dotaquery"
+
     const val MUTATION = "mutation"
+    const val ROOTMUTATIONTYPE = "rootmutationtype"
+    const val ROOTMUTATION = "rootmutation"
+    const val DOTAMUTATION = "dotamutation"
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/problem/graphql/GqlConst.kt:65]: please add some comments/documentation for these constants. eg, what is the difference between `QUERY_ROOT` and `ROOTQUERY`?

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/IntrospectiveQuery.kt [hunk 2]
```diff
@@ -70,6 +83,22 @@ class IntrospectiveQuery {
...
         val body = response.readEntity(String::class.java)
 
+        //TODO parse this body here to see if it has any "errors" field and no "data"
+
+        val jackson = ObjectMapper()
+
+        val node: JsonNode = try {
+            jackson.readTree(body)
+        } catch (e: JsonProcessingException) {
+            return null
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/problem/graphql/IntrospectiveQuery.kt:93]: this should not be ignored.
if what returned is not a JSON, should log it as an error and throw a `SutProblemException` with a meaningful error message

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/IntrospectiveQuery.kt [hunk 3]
```diff
@@ -70,6 +83,22 @@ class IntrospectiveQuery {
...
+        }
+
+        val withErrors= node.findPath("errors")
+
+       if (!withErrors.isEmpty){
+            throw SutProblemException("Failed to retrieve GraphQL schema: response contains error.")
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/problem/graphql/IntrospectiveQuery.kt:99]: add here the message of the error. could simply output `body`

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/GqlConst.kt [hunk 4]
```diff
@@ -49,7 +49,10 @@ object GqlConst {
...
     const val LIST = "list"
     const val INPUT_OBJECT = "input_object"
-    /**/
+    /*
+    *Those are entry points of GraphQL query and mutation
+    * Queries and mutations names are defined by developers, so they can have different names
+    * eg: query, rootquery ...  */
     const val QUERY = "query"
     const val QUERY_TYPE = "querytype"
     const val ROOT = "root"
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/problem/graphql/GqlConst.kt:58]: names could be anything. cannot hardcode here.
need to check based on actual type in schema, NOT the name.
see https://graphql.org/learn/schema/

**Review by arcuri82 — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/GqlConst.kt (hunk 4 above)
Comment (arcuri82) [reply core/src/main/kotlin/org/evomaster/core/problem/graphql/GqlConst.kt:58]: create a test in which we have something like `foo: Query`

Comment (asmab89) [reply core/src/main/kotlin/org/evomaster/core/problem/graphql/GqlConst.kt:58]: Done

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/GqlConst.kt (hunk 1 above)
Comment (arcuri82) [reply core/src/main/kotlin/org/evomaster/core/problem/graphql/GqlConst.kt:68]: did u add the documentation? or, if those are not used anymore, should be removed from the code

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLUtils.kt [hunk 5]
```diff
@@ -208,6 +208,8 @@ object GraphQLUtils {
      * graph: a map with the name of the node as a key and its fields and edges as values,
      * history: used in the cycles management,
      * objectFieldsHistory: a set used in the management of already constructed nodes
+
+     * TODO Remove isRoot(), and extract the names directly from the schema
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLUtils.kt:212]: is this TODO still valid?

**Review by arcuri82 — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLUtils.kt (hunk 5 above)
Comment (asmab89) [reply core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLUtils.kt:212]: Done.

**Pull request merged**